studio/hooks/misc/useSelectedOrganization.ts

```diff
@@ -3,15 +3,23 @@ import { useOrganizationsQuery } from 'data/organizations/organizations-query'
 import { useMemo } from 'react'
 
 import { useSelectedProject } from './useSelectedProject'
+import { LOCAL_STORAGE_KEYS } from 'lib/constants'
 
 export function useSelectedOrganization({ enabled = true } = {}) {
   const { slug } = useParams()
   const { data } = useOrganizationsQuery({ enabled })
   const selectedProject = useSelectedProject()
+  const localStorageSlug =
+    typeof window !== 'undefined'
+      ? localStorage.getItem(LOCAL_STORAGE_KEYS.RECENTLY_VISITED_ORGANIZATION)
+      : undefined
```

Comment:
Do we want to consider caching this in a valtio store to prevent slowing down react's rendering?

Reply:
just to check though cause useSelectedOrganization already returns a useMemo - would this already mitigate the performance issue here?